pkg/services/librarypanels/database_test.go

```diff
@@ -0,0 +1,82 @@
+package librarypanels
+
+import (
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddLibraryPanel(t *testing.T) {
+	t.Run("should fail if library panel already exists", func(t *testing.T) {
+		lps, user := setupTestEnv(t, models.ROLE_EDITOR)
+		command := addLibraryPanelCommand{
+			OrgId:        1,
+			FolderId:     1,
+			SignedInUser: &user,
+			Title:        "Text - Library Panel",
+			Model: []byte(`
+	{
+      "datasource": "${DS_GDEV-TESTDATA}",
+      "id": 1,
+      "title": "Text - Library Panel",
+      "type": "text"
+    }
+`),
+		}
+
+		noErr := lps.addLibraryPanel(&command)
+		require.NoError(t, noErr)
+
+		err := lps.addLibraryPanel(&command)
+		require.Error(t, err)
```

Comment:
Probably want `require.EqualError` here, to be precise about the expected error.

Reply:
Fixed

Reply:
It's still `Error` and not `EqualError`.

Reply:
Fixed :)